### PR TITLE
SLDEV-7680

### PR DESCRIPTION
**Pull request by bartoszkaczkowski**
- Adjusted to changes in pom integration: collecting pom files now results in `CollectedPomFiles` not in `List<PomFile>`, `backup` method now takes a workspace path as an argument (though the path is used only in a case unrelated to jenkins-plugin).

- Bumped up used infra major version to 3.

- Added exception to enforcer rule for bytecode version for jackson library. Jackson is compatible with java 1.7, but it's build with a plugin that also puts in its JAR a file used for Java 9 modules. The file's presence causes enforcer to fail the build.

### CODE REVIEW

File: src/main/java/io/sealights/plugins/sealightsjenkins/BeginAnalysis.java [hunk 1]
```diff
@@ -819,29 +820,24 @@ private boolean validateAndTryUseToken(Logger logger, String token, SeaLightsPlu
...
             }
         }
 
-
-        if (!isParentPomInList) {
-            pomFiles.add(new JenkinsPomFile(pomPath, logger));
-        }
-
-        return pomFiles;
+        return new CollectedPomFiles(new JenkinsPomFile(pomPath, logger), null, nonRootPomFiles);
```
Comment (RoniSegal) [inline src/main/java/io/sealights/plugins/sealightsjenkins/BeginAnalysis.java:840]: Why didnt you add `toList()` method and return a list? if so, you did not need to change the signature of the method

File: src/main/java/io/sealights/plugins/sealightsjenkins/BeginAnalysis.java [hunk 2]
```diff
@@ -819,29 +820,24 @@ private boolean validateAndTryUseToken(Logger logger, String token, SeaLightsPlu
         return true;
     }
 
-    private List<PomFile> getPomFiles(List<String> folders, String patterns, Logger logger, String pomPath) throws IOException, InterruptedException {
-        List<PomFile> pomFiles = new ArrayList<>();
-        boolean isParentPomInList = false;
```
Comment (RoniSegal) [inline src/main/java/io/sealights/plugins/sealightsjenkins/BeginAnalysis.java:824]: Why did you remove that flag?

**Review by RoniSegal — CHANGES_REQUESTED** [review]

File: src/main/java/io/sealights/plugins/sealightsjenkins/BeginAnalysis.java (hunk 1 above)
Comment (bartoszkaczkowski) [reply src/main/java/io/sealights/plugins/sealightsjenkins/BeginAnalysis.java:840]: because I need a `CollectedPomFiles` instance to pass to `MavenIntegrationInfo`'s constructor.

File: src/main/java/io/sealights/plugins/sealightsjenkins/BeginAnalysis.java (hunk 2 above)
Comment (bartoszkaczkowski) [reply src/main/java/io/sealights/plugins/sealightsjenkins/BeginAnalysis.java:824]: the point of this flag was to prevent putting duplicate `PomFile` of the root pom in the list. I took care of it differently (i.e. not putting the root pom in the list ever, and then creating the `PomFile` of the root in line 840.

**Review by RoniSegal — APPROVED** [review]

**Review by alasch2 — APPROVED** [review]
